serial/tools/list_ports_linux.py

```diff
@@ -90,14 +90,11 @@ def read_line(self, *args):
 
 def comports(include_links=False):
     devices = set()
-    devices.update(glob.glob('/dev/ttyS*'))     # built-in serial ports
-    devices.update(glob.glob('/dev/ttyUSB*'))   # usb-serial with own driver
-    devices.update(glob.glob('/dev/ttyXRUSB*')) # xr-usb-serial port exar (DELL Edge 3001)
-    devices.update(glob.glob('/dev/ttyACM*'))   # usb-serial with CDC-ACM profile
-    devices.update(glob.glob('/dev/ttyAMA*'))   # ARM internal port (raspi)
-    devices.update(glob.glob('/dev/rfcomm*'))   # BT serial devices
-    devices.update(glob.glob('/dev/ttyAP*'))    # Advantech multi-port serial controllers
-    devices.update(glob.glob('/dev/ttyGS*'))    # https://www.kernel.org/doc/Documentation/usb/gadget_serial.txt
+    drivers = open('/proc/tty/drivers').readlines()
```

Comment:
You're leaking the fd here:
```suggestion
    with open('/proc/tty/drivers') as drivers_file:
        drivers = drivers_file.readlines()
```

Reply:
Great! When can this be applied?